### PR TITLE
Remove horizontal lines at top of page

### PR DESCRIPTION
**Pull request by jsha**
They are not needed to separate the search bar and the title, which are visually distinct on their own.

Part of #59840

Demo: https://rustdoc.crud.net/jsha/fewer-lines/std/string/struct.String.html

r? @GuillaumeGomez

### CODE REVIEW

**Comment by camelid** [comment]
Ah, what a nice improvement!

File: src/test/rustdoc-gui/headings.goml
Comment (GuillaumeGomez) [inline src/test/rustdoc-gui/headings.goml:18]: I think you simply remove these lines. There is no point to check if something doesn't exist in this case.

**Comment by camelid** [comment]
@jsha the test failure was not from bors, it was from PR CI. You can re-run that by clicking "re-run all jobs" on this page: https://github.com/rust-lang/rust/runs/4793628088?check_suite_focus=true. But it might not be worth it.

@bors r-

**Comment by jsha** [comment]
Bump on this one when you get a chance @GuillaumeGomez.

**Comment by GuillaumeGomez** [comment]
@jsha I don't think it's coming from your PR but just in case, can you rebase it on top of master please? Then r=me.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Comment by jsha** [comment]
Thanks for the review! Temporarily bumping this out of the queue because it has a change to toggle-docs-mobile that will conflict with the changes in #92629. Letting that one merge first.

@bors r-

**Comment by camelid** [comment]
@jsha should this be re-approved now or does it need a new eview?

**Comment by jsha** [comment]
It shouldn't get reapproved quite yet. There's another PR in the queue (https://github.com/rust-lang/rust/pull/92861) that also touches toggle-docs-mobile and will generate a conflict. Once that's landed I'll rebase this and reapprove it.

**Comment by jsha** [comment]
@bors r=GuillaumeGomez rollup

I decided to make another revision on #92861, so it will need another review; this one can go ahead of it.

**Pull request merged**
